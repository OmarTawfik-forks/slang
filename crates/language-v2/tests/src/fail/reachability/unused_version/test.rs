#![allow(unused_crate_dependencies)]

language_v2_macros::compile!(Language(
    name = Foo,
    binding_rules_file = "bindings/rules.msgb",
    root_item = Foo,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0", "2.0.0", "3.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Struct(
                    name = Foo,
                    fields = (field_1 = Optional(reference = Bar, enabled = From("2.0.0")))
                ),
                Struct(name = Bar, fields = (field_1 = Optional(reference = Baz))),
                Token(name = Baz, definitions = [TokenDefinition(Atom("baz"))])
            ]
        )]
    )],
    built_ins = []
));

fn main() {}
