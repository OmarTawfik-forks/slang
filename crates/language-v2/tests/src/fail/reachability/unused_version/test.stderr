error: Item 'Bar' is not used in versions: 1.0.0..2.0.0
  --> src/fail/reachability/unused_version/test.rs:19:31
   |
19 |                 Struct(name = Bar, fields = (field_1 = Optional(reference = Baz))),
   |                               ^^^
